### DISCUSSION TITLE
Identify Widget close, back, end conversation events

### DISCUSSION THREAD

**Discussion by Arun-gox** [chatwoot/chatwoot#9733] [Feature Requests]
### Is your feature or enhancement related to a problem? Please describe.

When user closes or goes back in widget, we need to capture that event so that we can manipulate the toggle or iframe. Same for ending conversation as well

### Describe the solution you'd like

When the widget is open and if user clicks the close button, similar to "chatwoot:on-message" event we need a event listener.

**Comment by pranavrajs** [comment]
@Arun-gox Moving this to discussions. 

Can you please share the use case and the need for it?

**Comment by Arun-gox** [comment]
@pranavrajs 
We have implemented the widget in our support menu. Once user chooses to chat with support the widget opens and until the conversation ends we don't display other menu option.

We need to capture if the user closes the menu or ends a conversation, so that we can toggle our menu to show other options and hide the iframe

**Reply by pranavrajs** [reply]
Is it possible to share a screencast of the existing workflow?

**Reply by Arun-gox** [reply]
1. Initial support menu
![Screenshot from 2024-07-10 17-42-29](https://github.com/chatwoot/chatwoot/assets/118156785/787868d8-ebd6-4819-9530-2a905292d225)

2. On clicking chat with support
![Screenshot from 2024-07-10 17-41-56](https://github.com/chatwoot/chatwoot/assets/118156785/bde76290-25fc-43d7-81f7-cb06ba08604b)

After starting the conversation and user closes the widget we're unable to capture so we made sure if next time user clicks the support menu the Chatwoot widget open.
